modules/navier_stokes/src/userobjects/CNSFVLeastSquaresSlopeReconstruction.C

```diff
@@ -128,7 +128,7 @@ CNSFVLeastSquaresSlopeReconstruction::reconstructElementSlope()
 
     /// for internal side
 
-    if (elem->neighbor(is) != NULL)
+    if (elem->neighbor(is) != NULL && hasBlocks(elem->neighbor(is)->subdomain_id()))
```

Comment:
Notice that `elem->neighbor(is)` is obtained on the first line in the body of this if-statement. So let's do:
```
    if (elem->neighbor(is) != NULL)
    {
      const Elem * neig = elem->neighbor(is);
      if (this->hasBlocks(neig->subdomain_id()))
      {
        <original body>
      }
```